### PR TITLE
K8s client staging

### PR DESCRIPTION
**Pull request by prashantvicky**
Checking (StatefulSets part 2)
1) Added new file (KubernetesClientApiConverter) 
Contains Methods 
a) fromStatefulSet
b) fromContainer
c) fromVolume
d) fromV1Probe

@lwander Please Review 

FYI  @chlung 
Regards
Prashant

We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+import io.kubernetes.client.models.V1Volume
+import io.kubernetes.client.models.V1beta1StatefulSet
+import io.kubernetes.client.models.V1Handler
+import io.kubernetes.client.models.V1ExecAction
+import io.kubernetes.client.models.V1TCPSocketAction
+import io.kubernetes.client.models.V1HTTPGetAction
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.description.servergroup.*
+
+/**
+ * Created by spinnaker on 20/8/17.
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy]: Please remove

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+ */
+class KubernetesClientApiConverter {
+
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy]: Extra newline

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+          type: seLinuxOptions.type,
+          level: seLinuxOptions.level
+        )
+      }
+    }
+
+    containerDescription.livenessProbe = fromV1Probe(container?.livenessProbe)
+    containerDescription.readinessProbe = fromV1Probe(container?.readinessProbe)
+
+
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy]: Extra newlines

**Comment by prashantvicky** [comment]
1) feature(provider/kubernetes) : In order to see Statefulset in Spinnaker UI we need to have loadbalancer details also as its throw null pointer at com.netflix.spinnaker.clouddriver.controllers.ApplicationsController on line no: 101 (result.clusters[account] << new ApplicationClusterViewModel(name: cluster.name, loadBalancers: cluster.loadBalancers.name as TreeSet, serverGroups: cluster.serverGroups*.name as TreeSet, provider: cluster.type) at this call 'cluster.loadBalancers.name')
2) Added Persistance Volume Claim for statefulset.

@lwander Please review. 
@chlung

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy [hunk 4]
```diff
@@ -48,6 +49,7 @@ class DeployKubernetesAtomicOperationDescription extends KubernetesAtomicOperati
   String serviceAccountName
   Integer sequence
   KubernetesPodSpecDescription podSpec
+  List<V1PersistentVolumeClaim> volumeClaimList
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/description/servergroup/DeployKubernetesAtomicOperationDescription.groovy]: I don't think this belongs here. Every other subresource is defined within clouddriver

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+import io.kubernetes.client.models.V1Volume
+import io.kubernetes.client.models.V1beta1StatefulSet
+import io.kubernetes.client.models.V1Handler
+import io.kubernetes.client.models.V1ExecAction
+import io.kubernetes.client.models.V1TCPSocketAction
+import io.kubernetes.client.models.V1HTTPGetAction
+
+
+/**
+ * Created by spinnaker on 20/8/17.
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy]: Please remove

**Comment by prashantvicky** [comment]
1) Removed V1PersistentVolumeClaim (Added the right implementation) 
2) Removed extra Space
 @lwander  Please review , and if all gud please merge it.
FYI @chlung 
Thanks
Prashant

**Pull request merged**